File: src/Avalonia.Samples/CompleteApps/Avalonia.MusicStore/Models/Album.cs
namespace Avalonia.MusicStore.Models
{
    // We use a record for Album since it helps to encapsulate data. 
    // See: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record 
    public record Album(string Artist, string Title, string CoverUrl);
}

File: src/Avalonia.Samples/FSharp/CompleteApps/MusicStore/Models/Album.fs
namespace MusicStore.Models

/// Represents a music album with its core metadata.
[<AllowNullLiteral>]
type Album(artist: string, title: string, coverUrl: string) =
    member val Artist: string = artist with get, set
    member val Title: string = title with get, set
    member val CoverUrl: string = coverUrl with get, set
